src/auth0/client.js

```diff
@@ -43,19 +45,23 @@ function pagedManager(client, manager) {
             data.push(...getEntity(rsp));
             const total = rsp.total || 0;
             const pagesLeft = Math.ceil(total / perPage) - 1;
-
             // Setup pool to get the rest of the pages
             if (pagesLeft > 0) {
-              await client.pool.addEachTask({
+              const pages = await client.pool.addEachTask({
                 data: Array.from(Array(pagesLeft).keys()),
                 generator: (page) => {
-                  const pageArgs = [ ...newArgs ];
+                  const pageArgs = _.cloneDeep(newArgs);
                   pageArgs[0].page = page + 1;
-                  return target[fnName](...pageArgs).then((r) => {
-                    data.push(...getEntity(r));
-                  });
+
+                  return target[fnName](...pageArgs).then(r => getEntity(r));
                 }
-              }).promise().then(results => data.push(...flatten(results)));
+              }).promise().then(results => flatten(results));
```

Comment:
nit: since you're awaiting already, you could move this flatten outside the promise chain as well